### PR TITLE
update for react 16 and add unit tests

### PR DESCRIPTION
**Pull request by duckworth**
I updated to work with react 16 and added unit tests. I also allow the style passed in to override the predefined style.

### CODE REVIEW

**Comment by isaachinman** [comment]
Nice! I will have to find some time to look through this, especially this time of year. If all is good, hopefully we can merge this and have it published on npm within a week or two.

File: setupTests.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import Enzyme from 'enzyme'
+import Adapter from 'enzyme-adapter-react-16';
+import chai from 'chai'
+import chaiEnzyme from 'chai-enzyme'
+
+chai.use(chaiEnzyme())
+Enzyme.configure({ adapter: new Adapter() })
```
Comment (isaachinman) [inline setupTests.js:7]: I'm not sure this is entirely necessary. Can we do something a bit simpler, closer to [this example](https://github.com/facebook/jest/tree/master/examples/react) from Facebook?

**Review by isaachinman — CHANGES_REQUESTED** [review]
Looks good altogether. Just wondering if we can simplify the jest setup.

Also just want to note that I am going to change a few things prior to a v1.1.0 release regarding babel and deps to align with [this boilerplate](https://github.com/isaachinman/javascript-project-boilerplate) that I use for JavaScript projects.

Thanks for taking an interest in this little project!

File: setupTests.js (hunk 1 above)
Comment (duckworth) [reply setupTests.js:7]: I included chai-enzyme  for the style assertions "to.have.style" as it seemed easiest.

I started with that approach and then saw that react-addons-test-utils was deprecated and when I went to https://reactjs.org/docs/test-utils.html to see what was needed to migrate they recommended Enzyme. I also needed to use mount from Enzyme for Full Dom Rendering and it seemed cleaner than TestUtils.renderIntoDocument(). 

I am open to suggestions but it I managed to get this working and it took longer than I expected already.

Comment (isaachinman) [reply setupTests.js:7]: Okay, no problem. I'll merge this now and publish it on npm. Sometime in the next few weeks I'll refactor and publish again. Cheers for the work and happy holidays!

**Pull request merged**